### PR TITLE
Refactor statefulset test with sets.String

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
Delete redundant sort. These string slices only own one element.
There is no necessary to sort them.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhangxiaoyu-zidif** [comment]
@foxish 
PTAL

**Comment by enisoc** [comment]
I would actually prefer not to remove these sorts, because the cost is not significant for a test, yet they provide protection in case someone alters the test to add a second item, or copies the code for another test case.

Ideally this should be changed to make `got` and `want` use [sets.String](https://godoc.org/github.com/kubernetes/apimachinery/pkg/util/sets#String) instead, since that is both safe and more clearly shows the intent. Would you be willing to repurpose this PR to do that?

**Comment by zhangxiaoyu-zidif** [comment]
@enisoc
yes， I like sets.String. I fix it with set.String.
Should it be necessary to change slice to set.String in other codes?
If so, could I assign this kind of PR to you?

**Comment by enisoc** [comment]
This looks good. Could you make the whole file consistent so this PR is self-contained?

**Comment by zhangxiaoyu-zidif** [comment]
@xiangpengzhao 
Hi, peter, please /ok-to-test. 
I need to see if it has build error.

**Comment by zhangxiaoyu-zidif** [comment]
@Hi, peter
It seems that fixing BUILD manully does not work.

File: pkg/controller/statefulset/BUILD [hunk 1]
```diff
@@ -74,6 +74,7 @@ go_test(
         "//vendor/k8s.io/apimachinery/pkg/labels:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/runtime:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/types:go_default_library",
+	"//vendor/k8s.io/apimachinery/pkg/util/sets:go_default_library",
```
Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/BUILD]: This line seems have one more space.

**Comment by zhangxiaoyu-zidif** [comment]
@xiangpengzhao 
My bad. I use "tab" mistakenly...
eight spaces is right...

**Comment by xiangpengzhao** [comment]
OK. It would be safe to let `hack/update-bazel.sh` update it. Try to fix the script issue.

**Comment by zhangxiaoyu-zidif** [comment]
Alright.I need dig the reason out...

**Comment by xiangpengzhao** [comment]
This is a proxy issue. Seems like Gavin has a workaround. @guangxuli

**Comment by zhangxiaoyu-zidif** [comment]
@xiangpengzhao which proxy issue?

**Comment by zhangxiaoyu-zidif** [comment]
@enisoc 
I checked the whole file. And fix the other one. PTAL
Thanks.

**Comment by guangxuli** [comment]
@zhangxiaoyu-zidif In my workground it is related to network, just set **http_proxy="your proxy"** before running the script. Any way, must make the proxy configuration valid in `hack/lib/util.sh`

File: pkg/controller/statefulset/stateful_set_test.go [hunk 2]
```diff
@@ -521,16 +521,13 @@ func TestGetPodsForStatefulSetAdopt(t *testing.T) {
 	if err != nil {
 		t.Fatalf("getPodsForStatefulSet() error: %v", err)
 	}
-	var got []string
+	got := sets.NewString()
 	for _, pod := range pods {
-		got = append(got, pod.Name)
+		//got = append(got, pod.Name)
```
Comment (enisoc) [inline pkg/controller/statefulset/stateful_set_test.go]: Don't need this line anymore.

File: pkg/controller/statefulset/stateful_set_test.go [hunk 3]
```diff
@@ -521,16 +521,13 @@ func TestGetPodsForStatefulSetAdopt(t *testing.T) {
...
+		got.Insert(pod.Name)
 	}
-
-	// pod2 should be claimed, pod3 and pod4 ignored
-	want := []string{pod1.Name, pod2.Name}
-	sort.Strings(got)
-	sort.Strings(want)
-	if !reflect.DeepEqual(got, want) {
+	want := sets.NewString(pod1.Name, pod2.Name)
```
Comment (enisoc) [inline pkg/controller/statefulset/stateful_set_test.go]: Keep the comment explaining what's expected.

Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/stateful_set_test.go]: Fixed, Thanks a lot@enisoc

**Pull request merged**